framework/CodeInterfaces/SCALE/tritonAndOrigenData.py

```diff
@@ -0,0 +1,359 @@
+    """
+      Constructor
+      @ In, filen, string or dict, file name to be parsed or dictionary {'origen':filein1,'triton':filein2}
+      @ In, timeUOM, string, time uom (e.g. s, d, h, y)
+      @ In, outputType, string, output type (i.e. origen, triton or combined). If combined, the triton output (last step) is used as exported as initial condition of the origen one
+      @ Out, None
+    """
+    if not isinstance(filen,dict):
+      filenames = [filen]
+      outTypeDict = {filen:outputType}
+    else:
+      filenames = [filen[outputType]] if outputType != 'combined' else filen.values()
+      outTypeDict = dict(zip(filen.values(), filen.keys()))
+
+    # retrieve keff and kinf
+    self.data = {}
+    for outFile in filenames:
+      self.lines = open(os.path.abspath(os.path.expanduser(outFile)),"r").readlines()
+      if outTypeDict[outFile] == 'triton':
+        self.data['transportInfo'] = self.retrieveKeff()
+        if self.data['transportInfo'] is not None:
+          if not self.data['transportInfo']['timeUOM'].strip().startswith(timeUOM):
+            self.data['transportInfo']['time'] = (_scalingFactorBetweenTimeUOM(timeUOM, self.data['transportInfo']['timeUOM'])*np.asarray(self.data['transportInfo']['time'])).tolist()
+        # retrieve nuclide densities
+        self.data['nuclideDensities'] = self.retrieveNuclideConcentrations()
+        if self.data['nuclideDensities'] is not None:
+          if not self.data['nuclideDensities']['timeUOM'].strip().startswith(timeUOM):
+            self.data['nuclideDensities']['time'] = (_scalingFactorBetweenTimeUOM(timeUOM, self.data['nuclideDensities']['timeUOM'])*np.asarray(self.data['nuclideDensities']['time'])).tolist()
+        # retrieve mixture powers
+        self.data['mixPowers'] = self.retrieveMixtureInfo()
+        if self.data['mixPowers'] is not None:
+          if not self.data['mixPowers']['timeUOM'].strip().startswith(timeUOM):
+            self.data['mixPowers']['time'] = (_scalingFactorBetweenTimeUOM(timeUOM, self.data['mixPowers']['timeUOM'])*np.asarray(self.data['mixPowers']['time'])).tolist()
+      elif outTypeDict[outFile] == 'origen':
+        # origen history overview and concentration tables
+        self.data['histOverviewOrigen'] = self.retrieveHistoryOverview()
+        if self.data['histOverviewOrigen'] is not None:
+          self.data['concTablesOrigen'] = self.retrieveConcentrationTables()
+          if not self.data['concTablesOrigen']['timeUOM'].strip().startswith(timeUOM):
+            self.data['concTablesOrigen']['time'] = (_scalingFactorBetweenTimeUOM(timeUOM, self.data['concTablesOrigen']['timeUOM'])*np.asarray(self.data['concTablesOrigen']['time'])).tolist()
+          if not self.data['histOverviewOrigen']['timeUOM'].strip().startswith(timeUOM):
+            self.data['histOverviewOrigen']['time'] = (_scalingFactorBetweenTimeUOM(timeUOM, self.data['histOverviewOrigen']['timeUOM'])*np.asarray(self.data['histOverviewOrigen']['time'])).tolist()
```

Comment:
there are a lot if statements but much else ones: is this done on purpose?

Reply:
yes because all the blocks are optionals